cmd/pmm-agent-entrypoint/main.go

```diff
@@ -0,0 +1,267 @@
+// pmm-agent
+// Copyright 2019 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	reaper "github.com/ramr/go-reaper"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+	"gopkg.in/alecthomas/kingpin.v2"
```

Comment:
@BupycHuk Can we start using cobra here? The comparison doesn't look promising for kingpin https://go.libhunt.com/compare-kingpin-vs-cobra

Reply:
We use kingpin everywhere in PMM. If we want to use cobra we need to migrate pmm-agent entirely. We want to migrate to a monorepo, it seems not the best idea to add new dependencies now.

Reply:
How can you migrate then a component that has not been maintained for 4 years? What strategy do you suggest? Never is not a strategy I'm afraid.

Reply:
Not as part of this PR, we started using kong in mongodb_exporter and if we decide to migrate to another lib we should make a decision what tool will be used and will it be backward compatible with the kingpin.

What do you mean by saying how can we migrate component that hasn't been maintained? We will migrate pmm-agent which has kingpin as dependency and based on current state of project it does everything required from it, so it's not a blocker for migration.

Reply:
Got it, thanks!